indra/newview/llviewermessage.cpp

```diff
@@ -3139,87 +3139,77 @@ void process_crossed_region(LLMessageSystem* msg, void**)
 }
 
 
-
-// Sends avatar and camera information to simulator.
-// Sent roughly once per frame, or 20 times per second, whichever is less often
-
-const F32 THRESHOLD_HEAD_ROT_QDOT = 0.9997f;    // ~= 2.5 degrees -- if its less than this we need to update head_rot
-const F32 MAX_HEAD_ROT_QDOT = 0.99999f;         // ~= 0.5 degrees -- if its greater than this then no need to update head_rot
-                                                // between these values we delay the updates (but no more than one second)
-
```

Comment:
The logic around when to send an AgentUpdate message was split between llappviewer.cpp::idle() and this file.  I'm moving all of the decision logic out of llappviewer.cpp and into the implementation of send_agent_update().  Basically the logic can be summed up as: **send if we need a keep-alive message OR (we are not throttled AND something changed)**.